addOns/retest/src/main/java/org/zaproxy/addon/retest/ExtensionRetest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2014 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.retest;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.addon.automation.AutomationAlertTest;
+import org.zaproxy.addon.automation.AutomationPlan;
+import org.zaproxy.addon.automation.ExtensionAutomation;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionRetest extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionRetest";
+
+    private RetestMenu retestMenu;
+    private ZapMenuItem menuItemRetest;
+    private RetestAPI retestApi;
```

Comment:
This can just be local to the hook method, I believe that's the only place it's used.